File: app/src/main/java/ir/ghararemaghzha/game/fragments/ProfileFragment.java
package ir.ghararemaghzha.game.fragments;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.view.WindowManager;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.card.MaterialCardView;
import com.google.android.material.textview.MaterialTextView;
import com.tmall.ultraviewpager.UltraViewPager;
import com.tmall.ultraviewpager.transformer.UltraDepthScaleTransformer;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

import io.realm.Realm;
import ir.ghararemaghzha.game.R;
import ir.ghararemaghzha.game.adapters.ProfileViewPager;
import ir.ghararemaghzha.game.classes.MySharedPreference;
import ir.ghararemaghzha.game.dialogs.UserDetailsDialog;
import ir.ghararemaghzha.game.models.ProfileModel;
import ir.ghararemaghzha.game.models.QuestionModel;

import static ir.ghararemaghzha.game.classes.Const.GHARAREHMAGHZHA_BROADCAST_REFRESH;

public class ProfileFragment extends Fragment {

    private NavController navController;
    private Context context;
    private FragmentActivity activity;
    private MaterialCardView buy, edit, stat, scoreHelper;
    private View v;

    private Realm db;

    private final BroadcastReceiver br = new BroadcastReceiver() {
        @Override
        public void onReceive(Context context, Intent intent) {
            updateUI();
        }
    };

    public ProfileFragment() {
    }

    @Override
    public void onResume() {
        super.onResume();
        context.registerReceiver(br, new IntentFilter(GHARAREHMAGHZHA_BROADCAST_REFRESH));
    }

    @Override
    public void onPause() {
        super.onPause();
        context.unregisterReceiver(br);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        v = inflater.inflate(R.layout.fragment_profile, container, false);
        context = getContext();
        activity = getActivity();
        init();

        return v;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        navController = Navigation.findNavController(view);
    }

    private void init() {
        ((MaterialTextView) activity.findViewById(R.id.toolbar_title)).setText(R.string.profile_title);

        db = Realm.getDefaultInstance();


        buy = v.findViewById(R.id.profile_buy);
        edit = v.findViewById(R.id.profile_edit);
        stat = v.findViewById(R.id.profile_stat);
        scoreHelper = v.findViewById(R.id.profile_scorehelper);

        updateUI();

        onClicks();
    }

    private void initViewPager(List<ProfileModel> data) {
        UltraViewPager ultraViewPager = v.findViewById(R.id.profile_recycler);
        ultraViewPager.setScrollMode(UltraViewPager.ScrollMode.HORIZONTAL);
        ultraViewPager.setPageTransformer(true, new UltraDepthScaleTransformer());
        ultraViewPager.setMultiScreen(0.6f);
        ultraViewPager.setItemRatio(1.0f);
        ultraViewPager.setRatio(1.6f);
      //  ultraViewPager.setMaxHeight(800);
     //   ultraViewPager.setAutoMeasureHeight(true);
        ultraViewPager.setInfiniteLoop(true);
        ultraViewPager.setAdapter(new ProfileViewPager(context,data));
        ultraViewPager.setAutoScroll(4000);


    }

    private void updateUI() {
        List<ProfileModel> data = new ArrayList<>();
        data.add(new ProfileModel(
                context.getString(R.string.avatar_url, MySharedPreference.getInstance(activity).getUserAvatar()),
                MySharedPreference.getInstance(context).getUsername(),
                "امتیاز من: "+MySharedPreference.getInstance(context).getScore()));
        int passed = Integer.parseInt(MySharedPreference.getInstance(context).getDaysPassed());

        if (passed < 0) {

            data.add(new ProfileModel(R.drawable.profile_time,
                    context.getString(R.string.profile_time_card_tostart),
                    context.getString(R.string.profile_time, String.valueOf(Math.abs(passed)))));


        } else {
            ProfileModel model = new ProfileModel(R.drawable.profile_time,context.getString(R.string.profile_time_card),"");
            if (passed == 9) {
                model.setSubtitle(context.getString(R.string.profile_time_lastday));
            }else if (passed >= 10) {
                model.setSubtitle(context.getString(R.string.profile_time_end));
            }else {
                model.setSubtitle(context.getString(R.string.profile_time, String.valueOf(10 - passed)));
            }
            data.add(model);
        }


        data.add(new ProfileModel(R.drawable.profile_total,"سوالات این دوره",
                String.valueOf(db.where(QuestionModel.class).findAll().size())));

        data.add(new ProfileModel(R.drawable.profile_remain,"سوالات من",
                String.valueOf(db.where(QuestionModel.class)
                        .equalTo("bought",true)
                        .and().equalTo("userAnswer", "-1").findAll().size())));

        initViewPager(data);

    }

    private void onClicks() {
        buy.setOnClickListener(v ->
                navController.navigate(R.id.action_menu_profile_to_menu_buy)
        );

        edit.setOnClickListener(v -> navController.navigate(R.id.action_global_profileEditFragment));

        stat.setOnClickListener(v -> showDetailsDialog(MySharedPreference.getInstance(context).getUserId()));
        
        scoreHelper.setOnClickListener(v -> navController.navigate(R.id.action_global_scoreHelperFragment));
    }

    private void showDetailsDialog(String userId) {
        UserDetailsDialog dialog = new UserDetailsDialog(activity, userId);
        dialog.setCanceledOnTouchOutside(true);
        dialog.setCancelable(true);
        Objects.requireNonNull(dialog.getWindow()).setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        dialog.getWindow().getAttributes().windowAnimations = R.style.DialogAnimation;
        dialog.getWindow().setGravity(Gravity.CENTER);
        dialog.show();
        Window window = dialog.getWindow();
        window.setLayout(WindowManager.LayoutParams.MATCH_PARENT, WindowManager.LayoutParams.WRAP_CONTENT);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        if (db != null) db.close();
    }
}

File: app/src/main/java/ir/ghararemaghzha/game/fragments/ProfileFragment.kt
package ir.ghararemaghzha.game.fragments

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.*
import androidx.fragment.app.Fragment
import androidx.navigation.findNavController
import com.google.android.material.textview.MaterialTextView
import com.tmall.ultraviewpager.UltraViewPager
import com.tmall.ultraviewpager.transformer.UltraDepthScaleTransformer
import io.realm.Realm
import io.realm.kotlin.where
import ir.ghararemaghzha.game.R
import ir.ghararemaghzha.game.adapters.ProfileViewPager
import ir.ghararemaghzha.game.classes.Const.GHARAREHMAGHZHA_BROADCAST_REFRESH
import ir.ghararemaghzha.game.classes.MySharedPreference
import ir.ghararemaghzha.game.databinding.FragmentProfileBinding
import ir.ghararemaghzha.game.dialogs.UserDetailsDialog
import ir.ghararemaghzha.game.models.ProfileModel
import ir.ghararemaghzha.game.models.QuestionModel
import java.util.*
import kotlin.math.abs

class ProfileFragment : Fragment(R.layout.fragment_profile) {

    private lateinit var b: FragmentProfileBinding

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        b = FragmentProfileBinding.inflate(layoutInflater, container, false)
        return b.root
    }

    private val br: BroadcastReceiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context, intent: Intent) {
            updateUI()
        }
    }

    override fun onResume() {
        super.onResume()
        requireContext().registerReceiver(br, IntentFilter(GHARAREHMAGHZHA_BROADCAST_REFRESH))
    }

    override fun onPause() {
        super.onPause()
        requireContext().unregisterReceiver(br)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        init()
    }

    private fun init() {
        requireActivity().findViewById<MaterialTextView>(R.id.toolbar_title).setText(R.string.profile_title)
        updateUI()
        onClicks()
    }

    private fun onClicks() {
        b.profileBuy.setOnClickListener { requireView().findNavController().navigate(R.id.action_menu_profile_to_menu_buy) }
        b.profileEdit.setOnClickListener { requireView().findNavController().navigate(R.id.action_global_profileEditFragment) }
        b.profileStat.setOnClickListener { showDetailsDialog(MySharedPreference.getInstance(requireContext()).getUserId()) }
        b.profileScorehelper.setOnClickListener { requireView().findNavController().navigate(R.id.action_global_scoreHelperFragment) }
    }

    private fun showDetailsDialog(userId: String) {
        val dialog = UserDetailsDialog(requireActivity(), userId)
        dialog.setCanceledOnTouchOutside(true)
        dialog.setCancelable(true)
        dialog.window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
        dialog.window?.attributes?.windowAnimations = R.style.DialogAnimation
        dialog.window?.setGravity(Gravity.CENTER)
        dialog.show()
        val window = dialog.window
        window?.setLayout(WindowManager.LayoutParams.MATCH_PARENT, WindowManager.LayoutParams.WRAP_CONTENT)
    }

    private fun updateUI() {
        val db = Realm.getDefaultInstance()
        val data: MutableList<ProfileModel> = ArrayList()
        data.add(ProfileModel(
                getString(R.string.avatar_url, MySharedPreference.getInstance(requireContext()).getUserAvatar()),
                MySharedPreference.getInstance(requireContext()).getUsername(),
                "امتیاز من: " + MySharedPreference.getInstance(requireContext()).getScore()))
        val passed = MySharedPreference.getInstance(requireContext()).getDaysPassed()
        if (passed < 0) {
            data.add(ProfileModel(R.drawable.profile_time,
                    getString(R.string.profile_time_card_tostart),
                    getString(R.string.profile_time, abs(passed).toString())))
        } else {
            val model = ProfileModel(R.drawable.profile_time, getString(R.string.profile_time_card), "")
            when {
                passed == 6 -> model.subtitle = getString(R.string.profile_time_lastday)

                passed >= 7 -> model.subtitle = getString(R.string.profile_time_end)

                else -> model.subtitle = getString(R.string.profile_time, (7 - passed).toString())
            }

            data.add(model)
        }
        data.add(ProfileModel(R.drawable.profile_total, "سوالات این دوره", db.where<QuestionModel>().findAll().size.toString()))
        data.add(ProfileModel(R.drawable.profile_remain, "سوالات من", db.where<QuestionModel>().equalTo("userAnswer", "-1").findAll().size.toString()))
        initViewPager(data)
    }

    private fun initViewPager(data: List<ProfileModel>) {
        b.profileRecycler.setScrollMode(UltraViewPager.ScrollMode.HORIZONTAL)
        b.profileRecycler.setPageTransformer(true, UltraDepthScaleTransformer())
        b.profileRecycler.setMultiScreen(0.6f)
        b.profileRecycler.setItemRatio(1.0)
        b.profileRecycler.setInfiniteLoop(true)
        b.profileRecycler.adapter = ProfileViewPager(requireContext(), data)
        b.profileRecycler.setAutoScroll(4000)
    }

}